scala/integration/intellij-bsp/src/org/jetbrains/plugins/scala/bsp/server/connection/ScalaConnectionDetailsProvider.scala

```diff
@@ -0,0 +1,69 @@
+package org.jetbrains.plugins.scala.bsp.server.connection
+
+import ch.epfl.scala.bsp4j.BspConnectionDetails
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import kotlin.coroutines.Continuation
+import org.jetbrains.plugins.bsp.extension.points.BuildToolId
+import org.jetbrains.plugins.bsp.server.connection.ConnectionDetailsProviderExtension
+import org.jetbrains.plugins.scala.bsp.config.ScalaPluginConstants.BUILD_TOOL_ID
+
+import java.lang
+import java.util.concurrent.TimeUnit
+import scala.jdk.CollectionConverters._
+
+class ScalaConnectionDetailsProvider extends ConnectionDetailsProviderExtension {
+  override def onFirstOpening(project: Project, projectPath: VirtualFile, continuation: Continuation[_ >: lang.Boolean]): AnyRef = {
+    val connectionFile = getConnectionFile(projectPath)
+    if (connectionFile.isDefined) {
+      continuation.resumeWith(true)
+      return boolean2Boolean(true)
```

Comment:
ahh, so that's calling coroutines manually looks like :) but are you sure you're doing it correctly? I'm especially concerned about first calling resumeWith and then returning a value. [here](https://github.com/JetBrains/intellij-scala/blob/bff7c7fd6e83551c4c14c017c147790b8ee91ef6/scala/scala-impl/src/org/jetbrains/plugins/scala/startup/ProjectActivity.scala#L16) they seem to be doing it differently